api/google.go

```diff
@@ -63,3 +72,69 @@ func GetSCCHealthStatus(url string) int {
 	}
 	return 1
 }
+
+// GetSCCSourcesByName returns Security Command Center sources for given numeric orgID,
+// filtered by name by given regex
+// original: https://github.com/GoogleCloudPlatform/golang-samples/blob/master/securitycenter/findings/list_sources.go
+func GetSCCSourcesByName(orgID string, nameRegex string) (map[string]string, error) {
+	regex, err := regexp.Compile(nameRegex)
+	if err != nil {
+		return nil, errors.Wrap(err, "error compiling nameRegex")
+	}
+	// Instantiate a context and a security service client to make API calls.
+	ctx := context.Background()
+	client, err := securitycenter.NewClient(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "securitycenter.NewClient")
+	}
+	defer client.Close() // Closing the client safely cleans up background resources.
+
+	req := &securitycenterpb.ListSourcesRequest{
+		Parent: fmt.Sprintf("organizations/%s", orgID),
+	}
+	it := client.ListSources(ctx, req)
+	result := map[string]string{}
+	for {
+		source, err := it.Next()
+		if err == iterator.Done {
+			break
+		}
+		if err != nil {
+			return nil, errors.Wrap(err, "it.Next")
+		}
+
+		if match := regex.MatchString(source.DisplayName); match {
+			result[source.Name] = source.DisplayName
+		}
+	}
+	return result, nil
+}
+
+// GetSCCLatestEventTime return map of sources and their latest event update time difference with now
+// original: https://github.com/GoogleCloudPlatform/golang-samples/blob/master/securitycenter/findings/list_filtered_findings.go
+func GetSCCLatestEventTime(sources map[string]string) (map[string]time.Duration, error) {
+	result := make(map[string]time.Duration)
+	ctx := context.Background()
```

Comment:
Was any timeout meant here?

Reply:
Not originally, but it makes sense to do: I've added 20s timeout to both.